doc/tutorials/gapi/oak_devices/oak_devices.markdown

```diff
@@ -0,0 +1,38 @@
+Using DepthAI Hardware /OAK depth sensors {#tutorial_gapi_oak_devices}
+=======================================================================================
+
+@tableofcontents
+
+@prev_tutorial{tutorial_gapi_face_beautification}
+
+![hardwares](pics/oak.jpg)
+
+Depth sensors compatible with Luxonis DepthAI library are supported through OpenCV Graph API (or G-API) module. Depth map, RGB image and some other formats of output can be retrieved by using familiar interface of G-API module.
+
+In order to use DepthAI sensor with OpenCV you should do the following preliminary steps:
+-#  Install Intel DepthAI library/depthai-core (from here <https://github.com/luxonis/depthai-core>).
+
+-#  Configure OpenCV with DepthAI library support by setting WITH_OAK flag in CMake. If DepthAI library is found in install folders OpenCV will be built with depthai-core  (see a status WITH_OAK in CMake log).
+
+-#  Build OpenCV.
+
+Below an example to capture a frame and save in .png format:
+
+@code{.cpp}
```

Comment:
Normally we don't put code like this but use a `@snippet` macro instead -- so the code is taken from an actual compilable source.

Reply:
Thank you @dmatveev, I will still become a good contributor who follows good practices. I'm learning.